### PR TITLE
RATIS-1698. Add unit-test of listener related to leaderElection

### PR DESCRIPTION
**Pull request by codings-dan**
see https://issues.apache.org/jira/browse/RATIS-1698

### CODE REVIEW

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftConfigurationImpl.java [hunk 1]
```diff
@@ -231,6 +231,10 @@ boolean hasMajority(Collection<RaftPeerId> others, RaftPeerId selfId) {
         (oldConf == null || oldConf.hasMajority(others, selfId));
   }
 
+  long getMajorityCount() {
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftConfigurationImpl.java:234]: Use `int`.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/PeerConfiguration.java [hunk 2]
```diff
@@ -157,6 +157,10 @@ boolean hasMajority(Collection<RaftPeerId> others, RaftPeerId selfId) {
     return num > size() / 2;
   }
 
+  long getMajorityCount() {
```
[the remark this answers is not in the document]
Comment (codings-dan) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/PeerConfiguration.java]: done

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by codings-dan** [comment]
Is the environment unstable causing these test errors, these UT passes on my local build.
https://github.com/apache/ratis/runs/8247709789?check_suite_focus=true#step:5:631

**Comment by codings-dan** [comment]
Thanks @szetszwo for helping review the pull request!

**Pull request merged**